File: algorithms/171. Excel Sheet Column Number/solution.py
class Solution(object):
    def titleToNumber(self, s):
        """
        :type s: str
        :rtype: int
        """
        num = 0
        for c in s:
            num *= 26
            num += ord(c) - ord('A') + 1
        return num

# test
s = Solution().titleToNumber
assert 28 == s('AB')

File: algorithms/171. Excel Sheet Column Number/solution.cc
class Solution {
public:
    int titleToNumber(string s) {
        int n = 0;
        for ( char c : s ) {
            n = n * 26 + c - 'A' + 1;
        }
        return n;
    }
};

// 1000 / 1000 test cases passed.
// Status: Accepted
// Runtime: 6 ms
